src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/McpServer/McpServer-CSharp/Program.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Server;
+using System.ComponentModel;
+
+var builder = Host.CreateApplicationBuilder(args);
+
+if (args.Length == 0 || args[0] != "start-mcp")
+{
+    Console.Error.WriteLine("Error: invalid command. Use the 'start-mcp' command-line argument to start the MCP server.");
+    return 1;
+}
+
+var maxNumberEnv = Environment.GetEnvironmentVariable("MAX_RANDOM_NUMBER");
+if (!int.TryParse(maxNumberEnv, out var maxNumber) || maxNumber <= 0)
+{
+    Console.Error.WriteLine("Error: you must set the MAX_RANDOM_NUMBER environment variable to a positive integer.");
+    return 1;
+}
+
+builder.Logging.AddConsole(consoleLogOptions =>
+{
+    // Configure all logs to go to stderr (stdout is used for the MCP protocol messages).
+    consoleLogOptions.LogToStandardErrorThreshold = LogLevel.Trace;
+});
+
+// add the MCP services, discover tools from the current assembly
+builder.Services
+    .AddMcpServer()
+    .WithStdioServerTransport()
+    .WithTools<RandomNumberTool>();
+
+// add dependencies for the MCP server tools
+builder.Services.AddSingleton(_ => new RandomNumberService(maxNumber));
```

Comment:
Having a DI service feels unnecessary concept for this template. Simplify and just have the class/tool

Reply:
I would nix it -- we don't have similar patterns in simpler templates.  reduce concept count, just have the tools class without a service -- it doesn't add value to the template realistically nor the concepts for MCP.